streaming/src/test/scala/org/apache/spark/streaming/util/WriteAheadLogSuite.scala

```diff
@@ -58,49 +71,127 @@ class WriteAheadLogSuite extends SparkFunSuite with BeforeAndAfter {
     Utils.deleteRecursively(tempDir)
   }
 
-  test("WriteAheadLogUtils - log selection and creation") {
-    val logDir = Utils.createTempDir().getAbsolutePath()
+  test(testPrefix + "read all logs") {
+    // Write data manually for testing reading through WriteAheadLog
+    val writtenData = (1 to 10).map { i =>
+      val data = generateRandomData()
+      val file = testDir + s"/log-$i-$i"
+      writeDataManually(data, file)
```

Comment:
Actually, we dont guarantee cross release compatibility. So achieving that is less important. Its more important to actually test the base case with batching, write batched data and read batched data. Similar to the FileBasedWALSuite where this unit test tests the reader's ability to read manually/directly written data, i was expecting this same unit test for BatchedWALSuite to test reading of batched data by reading directly generated batched data.